### DISCUSSION TITLE
How to custom hc:MessageBox global?

### DISCUSSION THREAD

**Discussion by GF-Huang** [HandyOrg/HandyControl#821] [Q&A] [answered]
Hi, @ghost1372 
I want to set it `BorderThinkness` to 0. I had read this https://hosseini.ninja/handycontrol/extend_controls/messageBox/, but I don't like show message with `MessageBoxInfo` every time. Is it a global way to custom it?

And why the `hc:MessageBox` does not show in the completion list?

![image](https://user-images.githubusercontent.com/4510984/117927444-cc276980-b32c-11eb-8a41-3bf95e28cca7.png)

Second best, even if I write as the docs, my XAML designer will crash.

![image](https://user-images.githubusercontent.com/4510984/117933438-2677f880-b334-11eb-902d-0e568752ff54.png)

![image](https://user-images.githubusercontent.com/4510984/117933494-32fc5100-b334-11eb-8566-44fb5523e407.png)

Mini project to duplicate this issue. [MessageBoxDemo.zip](https://github.com/HandyOrg/HandyControl/files/6464504/MessageBoxDemo.zip)

**Comment by ghost1372** [comment, marked as answer]
you should overwrite messagebox style
It is definitely due to the weakness of the vs designer. There is no problem in runtime and the style is chosen correctly

**Reply by GF-Huang** [reply]
Yeah, but how to make it only effect runtime not design time? I want to keep designer works well.

**Reply by ghost1372** [reply]
🤔 I think you can inherit from control you want and in the Ctor check if DesignTime is true or not and if true return it.

```
public myControl()
{
if (DesignerProperties.GetIsInDesignMode(this))
{
return;
}
}

```

**Reply by GF-Huang** [reply]
What differences if return or not?

**Reply by ghost1372** [reply]
Well, if it return, it will not be rendered in design mode, but if it is not returned, the control will be rendered.
Return means that you no longer need to initialize the control and its codes

**Reply by GF-Huang** [reply]
Oh, you lost the `InitializeComponent()` call.

**Reply by ghost1372** [reply]
> Oh, you lost the `InitializeComponent()` call.

yeah 😅
